handlers/patchitem/patchItem.go

```diff
@@ -41,29 +40,38 @@ func (p *patchItem) Handle(request events.APIGatewayV2HTTPRequest) (interface{},
 		return nil, http.StatusBadRequest
 	}
 
-	newName, err := getName(request.Body)
+	newName, isCompleted, err := getFields(request.Body)
 	if err != nil {
 		log.Printf("Error: %s", err.Error())
 		return nil, http.StatusBadRequest
 	}
 
-	item, err := p.db.UpdateItem(listID, itemID, newName)
+	item, err := p.db.UpdateItem(listID, itemID, newName, isCompleted)
 	if err != nil {
 		if errors.Is(err, db.ErrorNotFound) {
 			return nil, http.StatusNotFound
 		}
+		if errors.Is(err, db.ErrorBadRequest) {
+			log.Printf("Error: %s", err.Error())
+			return nil, http.StatusBadRequest
+		}
 		log.Printf("Error: %s", err.Error())
 		return nil, http.StatusInternalServerError
 	}
 
 	return item, http.StatusOK
 }
 
-func getName(body string) (string, error) {
-	var item data.Item
-	err := json.Unmarshal([]byte(body), &item)
+func getFields(body string) (string, *bool, error) {
+	type Input struct {
+		Name        string `json:"Name"`
+		IsCompleted *bool  `json:"IsCompleted"`
+	}
```

Comment:
We can't re-use data.Item because we need to be able to tell the difference between `IsCompleted == false` and `IsCompleted == null` (i.e. unspecified)